aws_requests_auth/aws_auth.py

```diff
@@ -95,7 +100,7 @@ def __call__(self, r):
 
         # Create payload hash (hash of the request body content). For GET
         # requests, the payload is an empty string ('').
-        body = r.body if r.body else ''
+        body = r.body if r.body else bytes()
         payload_hash = hashlib.sha256(body).hexdigest()
```

Comment:
The key is that PreparedRequest.body is never a string, it is either None or binary data (bytes), encoding has already been applied by the time it becomes a request body. In PR #14 encoding is applied conditionally, but the only case it makes sense is when we replace None by an empty string. Instead of this hack we do the right thing and set body to an empty "bytes".